Sources_v3/StreamChat/Models/Message.swift

```diff
@@ -68,6 +68,17 @@ public struct _ChatMessage<ExtraData: ExtraDataTypes> {
     ///
     public let extraData: ExtraData.Message
     
+    // Workaround for storing values of the same type in struct.
+    var _quotedMessage: [_ChatMessage] = []
+    
+    /// Quoted message.
+    ///
+    /// If message is inline reply this property will contain message quoted by this reply.
+    ///
+    public var quotedMessage: _ChatMessage? {
+        _quotedMessage.first
+    }
```

Comment:
I'm not happy with this but the alternative is to switch to `class` that will introduce inconsistency in public facing API and need to provide initializer here as well as adjust Initializer from `MessageDTO` in `MessageDTO.swift`.
That's not really big deal but I just can't tell which option is better.

Reply:
I think this is actually dangerous in our use case. The backend makes sure the "quoted message depth" doesn't go more than 1 level but we can't do it locally. It means if we had a really long conversation with quoted replies we could end up wit serializing quite a lot of objects :) 
```
let originalMessageId = message.quotedMessage?.quotedMessage?.quotedMessage?.quotedMessage?.quotedMessage?.id
```

I think all we need to expose in this case is `quotedMessageId`. Users can easily fetch the data if needed via our `DataStore` API: 
```swift
let quotedMessage = client.dataStore.message(id: message.quotedMessageId!)
````